File: knowledge-base-backend-master/kb-user-auth/src/main/java/com/knowledge/base/userauth/service/TeamService.java
package com.knowledge.base.userauth.service;

import com.baomidou.mybatisplus.extension.service.IService;
import com.knowledge.base.common.result.PageResult;
import com.knowledge.base.userauth.dto.TeamCreateDTO;
import com.knowledge.base.userauth.dto.TeamQueryDTO;
import com.knowledge.base.userauth.dto.TeamUpdateDTO;
import com.knowledge.base.userauth.entity.Team;
import com.knowledge.base.userauth.vo.TeamMemberVO;
import com.knowledge.base.userauth.vo.TeamVO;

import java.util.List;

/**
 * Team Service interface
 *
 * @author airwzz999
 * @since 1.0.0
 */
public interface TeamService extends IService<Team> {

    /**
     * Create a team
     *
     * @param dto creation DTO
     * @return team ID
     */
    Long createTeam(TeamCreateDTO dto);

    /**
     * Update a team
     *
     * @param dto update DTO
     * @return whether successful
     */
    Boolean updateTeam(TeamUpdateDTO dto);

    /**
     * Delete a team
     *
     * @param teamId team ID
     * @return whether successful
     */
    Boolean deleteTeam(Long teamId);

    /**
     * Get team details
     *
     * @param teamId team ID
     * @return team VO
     */
    TeamVO getTeamDetail(Long teamId);

    /**
     * Paginated team query
     *
     * @param dto query DTO
     * @return paginated result
     */
    PageResult<TeamVO> pageTeams(TeamQueryDTO dto);

    /**
     * Get the team tree
     *
     * @param rootOnly whether to return only root teams (top-level teams)
     * @return team tree
     */
    List<TeamVO> getTeamTree(boolean rootOnly);

    /**
     * Add team members
     *
     * @param teamId  team ID
     * @param userIds user ID list
     * @return whether successful
     */
    Boolean addTeamMembers(Long teamId, List<Long> userIds);

    /**
     * Remove team members
     *
     * @param teamId team ID
     * @param userIds user ID list
     * @return whether successful
     */
    Boolean removeTeamMembers(Long teamId, List<Long> userIds);

    /**
     * Get team members
     *
     * @param teamId team ID
     * @return team member list (includes user information)
     */
    List<TeamMemberVO> getTeamMembers(Long teamId);
}

File: knowledge-base-frontend-master/src/services/team.service.ts
import { http } from './request';
import { Team, TeamMember } from '@/types';

export const teamService = {
  // Paginated query of teams
  getTeams: (params?: { current?: number; size?: number; teamName?: string; status?: number }) => {
    return http.post<{
      current: number;
      size: number;
      total: number;
      pages: number;
      records: Team[];
    }>('/auth/teams/page', params || {});
  },

  // Get team tree
  getTeamTree: (rootOnly = false) => {
    return http.get<Team[]>('/auth/teams/tree', { params: { rootOnly }, skipAuth: true } as any);
  },

  // Get team details
  getTeam: (id: string) => {
    return http.get<Team>(`/auth/teams/${id}`);
  },

  // Create team
  createTeam: (data: { teamName: string; teamCode: string; description?: string; icon?: string; leaderId: string; parentId?: string }) => {
    return http.post<number>('/auth/teams', data);
  },

  // Update team
  updateTeam: (data: { id: string; teamName?: string; teamCode?: string; description?: string; icon?: string; leaderId?: string; status?: number }) => {
    return http.put<boolean>('/auth/teams', data);
  },

  // Delete team
  deleteTeam: (id: string) => {
    return http.delete<boolean>(`/auth/teams/${id}`);
  },

  // Add team members (batch)
  addMembers: (teamId: string, userIds: string[]) => {
    return http.post<boolean>(`/auth/teams/${teamId}/members`, userIds);
  },

  // Remove team members (batch)
  removeMembers: (teamId: string, userIds: string[]) => {
    return http.delete<boolean>(`/auth/teams/${teamId}/members`, { data: userIds } as any);
  },

  // Get team member list
  getTeamMembers: (teamId: string) => {
    return http.get<TeamMember[]>(`/auth/teams/${teamId}/members`);
  },
};

export default teamService;
